assets/src/components/vehiclePropertiesCard.tsx

```diff
@@ -105,42 +107,53 @@ const VehicleWorkInfo = ({ vehicle }: VehicleProp): React.ReactElement => (
 // #endregion
 
 // #region Vehicle Location
-const CurrentLocation = ({
-  nearestIntersection,
-}: {
-  nearestIntersection?: string | null
-}): React.ReactElement => {
+const VehicleNearestIntersection = ({
+  vehicle,
+}: VehicleProp): React.ReactElement => {
+  const lastGoodResult = useRef<{
+    intersection: string
+    vehicleId: VehicleId
+  } | null>(null)
+  const statusResult = useNearestIntersectionResult(
+    vehicle.latitude,
+    vehicle.longitude
+  )
```

Comment:
**question:** is it possible to use `useState` rather than `useRef`?

Reply:
If so, I think `useState` would be preferable and a bit more idiomatic, `useRef` is mainly useful for imperatively accessing things in the DOM.

Reply:
IIRC I switched to `useRef` because it would cause infinite state update loops and the component does not need to be rerendered when this updates.

I can double check though.

I wanted to extract this into a wrapper around a hook that returns `LoadingStatus<Result<T, E>>` which would instead return `Cached<LoadingStatus<Result<T,E>>>` but left that up to the future rather than spending more time on it.

Reply:
https://reactjs.org/docs/hooks-reference.html#useref
> However, `useRef()` is useful for more than the `ref` attribute. It’s [handy for keeping any mutable value around](https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables) similar to how you’d use instance fields in classes.
>
> [...] Mutating the `.current` property doesn’t cause a re-render. [...]